pkg/informer/informer.go

```diff
@@ -244,140 +301,106 @@ func (d *DynamicDiscoverySharedInformerFactory) AddEventHandler(handler GVREvent
+	klog.V(2).InfoS("Determining dynamic informer additions and removals")
 
-		klog.Infof("Discovering types for logical cluster %q", logicalClusterName)
-		rs, err := d.disco.WithCluster(logicalcluster.New(logicalClusterName)).ServerPreferredResources()
-		if err != nil {
-			return err
+	gvrFor := func(group, version, resource string) schema.GroupVersionResource {
+		return schema.GroupVersionResource{
+			Group:    group,
+			Version:  version,
+			Resource: resource,
 		}
-		for _, r := range rs {
-			gv, err := schema.ParseGroupVersion(r.GroupVersion)
-			if err != nil {
-				return err
-			}
-			for _, ai := range r.APIResources {
-				gvr := gv.WithResource(ai.Name)
+	}
 
-				if strings.Contains(ai.Name, "/") {
-					// foo/status, pods/exec, namespace/finalize, etc.
-					continue
-				}
-				if !ai.Namespaced {
-					// Ignore cluster-scoped things.
-					continue
-				}
-				if !sets.NewString([]string(ai.Verbs)...).HasAll("list", "watch") {
-					klog.V(4).InfoS("resource is not list+watchable", "logical-cluster", logicalClusterName, "group", gv.Group, "version", gv.Version, "resource", ai.Name, "verbs", ai.Verbs)
-					continue
-				}
+	// Hard-code built in types that support list+watch
+	latest := map[schema.GroupVersionResource]struct{}{
+		gvrFor("", "v1", "configmaps"):                                                   {},
+		gvrFor("", "v1", "events"):                                                       {},
+		gvrFor("", "v1", "limitranges"):                                                  {},
+		gvrFor("", "v1", "namespaces"):                                                   {},
+		gvrFor("", "v1", "resourcequotas"):                                               {},
+		gvrFor("", "v1", "secrets"):                                                      {},
+		gvrFor("", "v1", "serviceaccounts"):                                              {},
+		gvrFor("certificates.k8s.io", "v1", "certificatesigningrequests"):                {},
+		gvrFor("coordination.k8s.io", "v1", "leases"):                                    {},
+		gvrFor("rbac.authorization.k8s.io", "v1", "clusterroles"):                        {},
+		gvrFor("rbac.authorization.k8s.io", "v1", "clusterrolebindings"):                 {},
+		gvrFor("rbac.authorization.k8s.io", "v1", "roles"):                               {},
+		gvrFor("rbac.authorization.k8s.io", "v1", "rolebindings"):                        {},
+		gvrFor("flowcontrol.apiserver.k8s.io", "v1beta2", "flowschemas"):                 {},
+		gvrFor("flowcontrol.apiserver.k8s.io", "v1beta2", "prioritylevelconfigurations"): {},
+		gvrFor("events.k8s.io", "v1", "events"):                                          {},
+		gvrFor("admissionregistration.k8s.io", "v1", "mutatingwebhookconfigurations"):    {},
+		gvrFor("admissionregistration.k8s.io", "v1", "validatingwebhookconfigurations"):  {},
+		gvrFor("apiextensions.k8s.io", "v1", "customresourcedefinitions"):                {},
```

Comment:
would like to see a unit test that compares this list to the server schema minus some disallow list.